rust/src/shards/gui/raw_context.rs

```diff
@@ -0,0 +1,254 @@
+use egui_gfx::egui_FullOutput;
+use egui_gfx::egui_Input;
+use egui_gfx::make_native_full_output;
+use egui_gfx::NativeFullOutput;
+
+use super::util;
+use super::EguiContext;
+use super::CONTEXTS_NAME;
+use super::EGUI_CTX_TYPE;
+use super::EGUI_UI_SEQ_TYPE;
+use super::GFX_CONTEXT_TYPE;
+use super::GFX_QUEUE_VAR_TYPES;
+use super::HELP_OUTPUT_EQUAL_INPUT;
+use super::INPUT_CONTEXT_TYPE;
+use super::PARENTS_UI_NAME;
+use crate::core::Core;
+use crate::shard::Shard;
+use crate::shardsc;
+use crate::shardsc::Shards;
+use crate::types::Context;
+use crate::types::ExposedInfo;
+use crate::types::ExposedTypes;
+use crate::types::InstanceData;
+use crate::types::OptionalString;
+use crate::types::ParamVar;
+use crate::types::Parameters;
+use crate::types::RawString;
+use crate::types::Seq;
+use crate::types::ShardsVar;
+use crate::types::Type;
+use crate::types::Var;
+use crate::types::WireState;
+use crate::types::ANY_TYPES;
+use crate::types::SHARDS_OR_NONE_TYPES;
+use std::ffi::CStr;
+
+struct UIContext {
+  context: Option<egui::Context>,
+  full_output: Option<NativeFullOutput>,
+  instance: ParamVar,
+  parents: ParamVar,
+  exposed: Vec<ExposedInfo>,
+}
+
+impl UIContext {
+  pub fn default() -> Self {
```

Comment:
Shouldn't it implement the `Default` trait instead?

```rust
impl Default for UIContext {
  fn default() -> Self;
}
```

Otherwise, call it `new()` to avoid confusion.